man/ola_recorder.1

```diff
@@ -32,6 +34,11 @@ Print
 version information
 .IP "--duration <uint32_t>"
 The length of time (seconds) to run for.
```

Comment:
As it's just tripped me up, can we reword this a bit to make it clear it's the total time it will playback for, what takes priority with loops and stop start etc.

Reply:
I'll work on it.  I also want to change this to milliseconds, as it's a bit confusing for this to be the only time argument that takes seconds, not milliseconds.

Reply:
Hmm, actually I'm not sure we should change it, as that breaks existing scripts...

Reply:
So initially I was sort of sold, but then I realised why I really don't like this. So the only "safe" way to do it would be to rename the argument, and then seamlessly auto-translate it and then throw an error if you try and use both with conflicting values.

To try and give an analogy, you setup a cron job running your ola_recorder apart from this one deals with money and pays you each time it's run, the payment amount is set to $1 (given you're American). An excellent programmer called Dan comes along and adds some more features to it, like tax calculations which take cents, he wants to standardise and so sets all the arguments to use cents. You see a new version and upgrade, next month you just get paid one cent a few months later you notice and you're rather upset.

That's why I'm reluctant. People don't read changelogs, despite our best efforts. We could compromise on a warning if your time is less than 2s say (i.e. you've fed in a time in seconds into a millisecond argument), which would at least make finding the cause of your issue easy, but it would be nice to get rid of it entirely.

I pondered floats/doubles, but it seems we're into another world of pain there if we want a certain amount of accuracy so still not ideal. Something decimal based feels like the right sort of way to go though anyway. Realistically the smallest time you want is 1ms, also most people want stuff to run for a number of seconds, and probably don't care about the ms, or only will in the sense of half a second. So if we moved to milliseconds we're also getting people to multiply by 1000 all the time.

Something decimal based feels like a win, a bit like this, but perhaps just simplified and as part of our string library:
https://github.com/vpiotr/decimal_for_cpp

Realistically, milliseconds is likely to be the main place we use it, or possibly the odd other one, so it could possibly just be a string split and then two strtol. I think we then win all round, people can keep just using 1s, or can enter 1.5/1.25/1.125 or whatever if they want a specific ms duration, but it's always clear what's going on. If you want it to run for hours or days, you don't end up with an unmanageable number of zeros after it either.

Thoughts? But like I say in the other one, shall we get the rest of it in (with seconds for the duration) and then look at this?

Reply:
I agree with you re. BC needs, just trying to gauge how best to handle it.  Sounds like the preferred solution is to leave duration as it was and handle the need for decimals across the code base at a later time.